uswds/uswds
# USWDS - Nav: Resolve padding overflow bug

# Summary

Fixed a bug that caused long nav button spans to overflow into its padding. Now, nav buttons properly break into two lines and maintain center alignment.

## Breaking change
This is not a breaking change.

## Related issue

Closes #5656

## Related pull requests

#5267
[Changelog →](https://github.com/uswds/uswds-site/pull/2387)

## Preview link

[All headers demo →](https://federalist-d5e7c07c-6ffa-4b8e-935f-49a7df24a505.sites.pages.cloud.gov/preview/uswds/uswds-sandbox/cm-demo-5655/)

[Header storybook preview →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-vertical-center-nav-padding-fix/?path=/story/components-header--default)

## Problem statement

When navigation button spans cause a line break, their text flows into their padding instead of breaking onto a new line

![image](https://github.com/uswds/uswds/assets/25211650/7334b444-6be3-4d22-b1ee-dbcb765d2eec)


## Solution

- Set the `span` element to `display: inline-block`
- Update `usa-nav__primary` child item flex alignment to `center`

## Testing and review

1. Visit the [testing repo](https://federalist-d5e7c07c-6ffa-4b8e-935f-49a7df24a505.sites.pages.cloud.gov/preview/uswds/uswds-sandbox/cm-demo-5655/) or [Header storybook preview](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-vertical-center-nav-padding-fix/?path=/story/components-header--default) story
2. In desktop views:
    1. Confirm larger spans do not overflow into padding area
    2. Confirm icons maintain center alignment
    3. Confirm nav buttons maintain equal size
    4. Confirm appropriate default and hover states
3. In mobile views:
    1. Confirm there is no visual regression
    2. Confirm icons are vertically aligned

### Testing checklist

- [ ]  Long nav button spans do not break center alignment
- [ ]  Buttons maintain equal size when one span breaks into two lines
- [ ]  Mobile styles are unaffected
- [ ]  Hover state is unaffected

## Further considerations

> [!NOTE]
> There seems to be a CSS behavior that causes a width calculation quirk and leaves empty space between the span and the icon. This behavior is present on `develop`.
>I briefly tried resolving but didn't find any quick solutions. Do we think this is out of scope or should we dig deeper?

![Screenshot 2023-12-05 at 1 39 36 PM](https://github.com/uswds/uswds/assets/25211650/4f6e1c3f-427d-420a-bd2d-e7dc465385a3)
![Screenshot 2023-12-05 at 2 20 21 PM](https://github.com/uswds/uswds/assets/25211650/f0d55a5a-dd03-4346-92d9-cace3ee93b96)
